src/nupic/frameworks/opf/opfutils.py

```diff
@@ -172,26 +183,34 @@ def isTemporal(inferenceType):
 
 
 
-# SensorInput - represents the mapping of a given inputRecord by the
-#   sensor region's encoder.
-#
-# dataRow:        A data row that is the sensor's "sourceOut" mapping of the
-#                 supplied inputRecord. The data row is a sequence of field
-#                 values that correspond to the schema returned by the
-#                 getDecodedFieldMetaInfo() method of the ModelIface-based
-#                 instance that returned this mapping.  See
-#                 ModelIface.getDecodedFieldMetaInfo() docstring for additional
-#                 details.
-#
-# dataEncodings:  A list of the corresponding bit-array encodings of each value
-#                 in "dataRow"
-#
-# sequenceReset:  The sensor's "resetOut" signal (0 or 1) emitted by the
-#                 sensor's compute logic on the supplied inputRecord; provided
-#                 for analysis and diagnostics.
-# TODO: document category
 
 class SensorInput(object):
+  """
+  Represents the mapping of a given inputRecord by the sensor region's encoder.
+
+  :param dataRow: A data row that is the sensor's "sourceOut" mapping of the
+                  supplied inputRecord. The data row is a sequence of field
+                  values that correspond to the schema returned by the
+                  getDecodedFieldMetaInfo() method of the ModelIface-based
+                  instance that returned this mapping.  See
+                  ModelIface.getDecodedFieldMetaInfo() docstring for additional
+                  details.
+
+  :param dataEncodings: A list of the corresponding bit-array encodings of each
+                  value in "dataRow"
+
+  :param sequenceReset:  The sensor's "resetOut" signal (0 or 1) emitted by the
+                  sensor's compute logic on the supplied inputRecord; provided
+                  for analysis and diagnostics.
+
+  :param dataDict:
+  :param category:
+
+  HELP: This ``dataRow`` description above is wrong because there is no
+        getDecodedFieldMetaInfo() method anymore. I aso need someone to doc
+        ``dataDict`` and ``category``. -- Matt
```

Comment:
It looks like this class is just an abstraction over the encoders. The `category` comes from `categoryOut` on the sensor region:

https://github.com/numenta/nupic/blob/914ada4fa719609e1523ca5c54c6e06b5b223bd3/src/nupic/frameworks/opf/htmpredictionmodel.py#L448

And in that code you can also see how `dataRow` comes from the sensor `sourceOut` while dataDict is the input to the sensor. Interestingly, these are both the unencoded values so I think they are just different forms of the same thing?

And the intent of this class isn't fully clear to me so hard to know the right thing to put in the documentation.